hls4ml/templates/vivado/nnet_utils/nnet_array.h

```diff
@@ -45,7 +45,7 @@ void transpose_3d(
                 idx_t[1] = idx[CONFIG_T::perm[1]];
                 idx_t[2] = idx[CONFIG_T::perm[2]];
 
-                data_t[idx_t[0] * dims_t[1] * dims_t[2] + idx_t[1] * dims_t[2] + idx_t[2]] = data[idx[0] * dims[1] * dims[2] + idx[1] * dims[2] + idx[2]];
+                data_t[idx_t[0] * dims_t[1] * dims_t[2] + idx_t[1] * dims_t[2] + idx_t[2]] = (res_T) data[idx[0] * dims[1] * dims[2] + idx[1] * dims[2] + idx[2]];
```

Comment:
It's related to the other comment, but somehow doing a cast here makes me nervous

Reply:
Right, I added the cast, but to clarify, I never expect data_T != res_T for the array version. Maybe, we should add an assert instead?

Reply:
removed cast